### ISSUE TITLE
Entering commands into terminal closes radar/cam view

### ISSUE THREAD

**Issue by Snooziepup** [darmuh/TerminalStuff#14]
Entering commands into the terminal when the monitor or cam views are onscreen causes them to disappear, being very inconvenient when the player wishes to close terminal doors and such.

**Comment by darmuh** [comment]
> Entering commands into the terminal when the monitor or cam views are onscreen causes them to disappear, being very inconvenient when the player wishes to close terminal doors and such.

Door codes shouldn't cause the monitor/cam views to disable from what I've seen in my dev testing. Do you mean the regular doors command and stuff like that? Currently it's pretty much intended for most commands to hide the monitor/cams views but if you have specific commands in particular that should not close it I can add those commands as exceptions.

**Comment by Snooziepup** [comment]
After some more testing, this only seems to happen with codes that are invalid, which will close the view. However if commands hiding the map view ect. are intended, I would love if an option or command toggle ect. was added to keep the view open when using commands as that's what I'm used to in regular gameplay, as I get to always keep a close eye on my crewmates even when doing other things.

**Comment by Snooziepup** [comment]
One other thing I've just found, while the "switch" command alone will not close the view, the command followed by a player name will, which is a little annoying as it's useful to use to switch to a specific player.

**Comment by darmuh** [comment]
> After some more testing, this only seems to happen with codes that are invalid, which will close the view. However if commands hiding the map view ect. are intended, I would love if an option or command toggle ect. was added to keep the view open when using commands as that's what I'm used to in regular gameplay, as I get to always keep a close eye on my crewmates even when doing other things.

Sure that's completely valid. My main reason for hiding cams views for other commands was mostly just because it looked ugly to me to see a load of white text load on top of the camera like when you type "help". I can see about adding a configuration that will allow for the vanilla-type behavior. I'm just not sure if my custom commands such as "vitals" will also allow for this. I'll update here after some testing. Also if you want to offer any further suggestions to expand on this mode please feel free 👍 

> One other thing I've just found, while the "switch" command alone will not close the view, the command followed by a player name will, which is a little annoying as it's useful to use to switch to a specific player.

I will have to look into this further. "switch" has definitely been the bane of my existence with these cams commands lol. I think while I have tied my logic into the regular "switch" command there may be another node for switching to particular players that I also need to hook into to. I'm going to look into it and see what I can do

**Comment by darmuh** [comment]
Solving this required completely reworking the switch, map, and cams commands. I can explain this in depth if anyone is interested but the summary of this is that the cams/maps commands will now work in the same way as the splitview commands (minimap/minicams).

I think I've also solved some of the lag between this mod and the fastplayerswitchview not detecting the switch command right away.

In short, there will be a new configurable: "camsNeverHide" which you will want to set to true. I've also updated the default so that the switch command or any interactions with the portable radar will not hide the cams views and feel somewhat seemless.

I also hope to have fixed a bug where the player name displayed on the terminal was not the correct one.

Will be pushing this in patch 2.1.1
